Add JMH benchmarking capabilities

burmanm opened the pull request:
This PR creates JMH based benchmarking setups to assist in finding the bottlenecks. Completely skips the REST-layer, JSON transformations and runtime (Wildfly), instead writing directly to the core-service.

The output looks like:

```
# Run complete. Total time: 00:09:58

Benchmark                              (datapointsPerMetric)  (size)  Mode  Cnt   Score   Error  Units
InsertBenchmark.insertBenchmark                            1  100000  avgt   10   4.489 ± 0.193   s/op
InsertBenchmark.insertBenchmark                           10  100000  avgt   10   8.655 ± 3.612   s/op
InsertBenchmark.insertBenchmarkSingle                      1  100000  avgt   10   8.881 ± 2.360   s/op
InsertBenchmark.insertBenchmarkSingle                     10  100000  avgt   10  11.152 ± 2.108   s/op
```

The Scassandra implementation and Mockito implementations do not work. The Scassandra throws some Scala induced issues at this point (Akka actor stuff) and Mockito requires more mocks to get MetricsService to understand findAllTenantIds and such methods (they cause NPE at this point).

burmanm commented:
Additional note, although the error looks very high (to make results unreliable), this is not the case in reality. It's because some Cassandra runs have some GC issues. An example of such run:

```
 Warmup Iteration   2: 8.669 s/op
# Warmup Iteration   3: 8.349 s/op
# Warmup Iteration   4: 13.901 s/op
# Warmup Iteration   5: 8.451 s/op
Iteration   1: 8.267 s/op
Iteration   2: 8.354 s/op
Iteration   3: 8.482 s/op
Iteration   4: 13.295 s/op
Iteration   5: 8.263 s/op
Iteration   6: 8.106 s/op
Iteration   7: 8.432 s/op
Iteration   8: 8.290 s/op
Iteration   9: 8.707 s/op
Iteration  10: 8.615 s/op
```

As you can see, only one benchmark iteration had that issue, but it's enough to cause large std error.

stefannegrea commented:
Do you mind moving this new module under integration-tests module and I will merge the PR afterwards.

burmanm commented:
Sure, I'll do that.

burmanm commented:
Modified, moved to under integration-tests and compiling is activated against a profile ``benchmark``, renamed these benchmarks to jmh-benchmark and allow them to accept configuration parameters from command line (to set Cassandra nodes and keyspace)

stefannegrea commented:
Thanks @burmanm!
